driver/utils/utils.h

```diff
@@ -179,8 +181,35 @@ inline SQLRETURN fillOutputBuffer(
 
     const auto bytes_to_copy = std::min<std::int64_t>(in_value_length, out_value_max_length);
     std::memcpy(out_value, in_value, bytes_to_copy);
+    
+    if (bytes_to_copy > out_value_max_length)
+        return SQL_SUCCESS_WITH_INFO;
```

Comment:
Why not just throw an exception here?

Reply:
Some processing later done in `fillOutputString()` (namely, null-terminating, if requested), so I would need to try-catch-retrhow in that case. That would be both slow, and hard to read/understand/follow code.

Reply:
(Even when truncated, strings have to be null-terminated.)